streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -384,6 +387,13 @@ public String queryableStoreName() {
     private SuppressedInternal<K> buildSuppress(final Suppressed<? super K> suppress) {
         if (suppress instanceof FinalResultsSuppressionBuilder) {
             final long grace = findAndVerifyWindowGrace(streamsGraphNode);
+            if (grace > Duration.ofHours(1).toMillis()) {
+                LOG.warn("Adding suppression to a windowed stream with a suspiciously long grace period of [{}]. " +
+                             "Suppressed results are delayed until the window closes and the grace period elapses. " +
+                             "If you wish to see fresher results, you can set it on the window definition using " +
```

Comment:
nit: `If you wish to see fresher results, you can set it...` -> `If you wish to see fresher results, you can set it to a lower value..`

Reply:
```suggestion
                             "If you wish to see fresher results, you can set it to a lower value on the window definition using " +
```